FileMapper/find_driveid_api.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * Tester function
+ */
+function testerFunction() {
+  var path = "C:\\Documents\\Muskan\\sheet1.xls";
+  Logger.log(findDriveMapping(path));
+}
+
+/**
+ * Finds the Drive Mapping for the given local 
+ * mapping using the config data
+ *
+ * @param {String} localPath Local Path whose mapping 
+ * is to be found
+ * @return {String} driveMapping The Drive id of the 
+ * mapped file/folder if found, otherwise it contains
+ * "Mapping Not Found, Ask the User".
+ */
+function findDriveMapping(localPath) {
+  var path = localPath;
+
+  // replaced back slash with forward slash
+  // to handle Mac file path
+  path = path.replace(/\//g, "\\");
+
+  var relativePath = "";
+  var found = false;
+  var driveMapping = "Mapping Not Found, Ask the User";
+
+  while (!found && path.length > 0) {
+    if (checkInConfig(path)) {
+      if (relativePath.length > 0) {
+        var currentDirectoryId = getFromConfig(path);
+        var currentDirectory = DriveApp.getFolderById(currentDirectoryId);
+        var relativeMapping = findInDrive(currentDirectory, relativePath);
+
+        if (relativeMapping !== null) {
+          addNewMappingToConfig(localPath, relativeMapping);
+          found = true;
+          driveMapping = relativeMapping;
+        }
+      }
+      else {
+        found = true;
+        driveMapping = getFromConfig(path);
+      }
+    }
+
+    if (found) {
+      break;
+    }
+
+    var position = path.lastIndexOf("\\");
```

Comment:
Would prefer if we write a Path util that takes care of all this irrespective of windows / unix path

Reply:
Done